virttest/utils_test/qemu.py

```diff
@@ -1842,3 +1842,308 @@ def get_all_memorys(self, vm):
+                                           usb_dev_params.get("usb_controller"),
+                                           usb_dev_params.get("port")):
+                if mig_protocol:
+                    tmp = copy.deepcopy(d)
+                    newdevs.append(tmp)
+                devs.insert(d)
+        vm.devices = devs
+        vm.create()
+        return newdevs
+
+
+class UsbStorageTest(UsbDevTest):
+
+    def __init__(self, test, params, env):
+        super(UsbStorageTest, self).__init__(test, params, env)
+
+    def get_usb_disk_list(self, session):
+
+        """
+        get the path of usb disk
+        """
+        list_disks = "readlink -f `ls /dev/disk/by-path/*|grep usb`"
+        output = session.cmd_output(list_disks).strip()
+        devlist = re.findall("/dev/sd\w", output)
+        if devlist:
+            return devlist
+
+    def usb_disk_io(self, with_fs, devname, session):
+
+        """
+        usb disk io test
+        """
+        if with_fs == "no":
+            write_cmd = "dd if=/dev/zero of=%s bs=1M count=100"
+            write_cmd = write_cmd % devname
+            read_cmd = "dd if=%s of=/dev/null bs=1M count=100"
+            read_cmd = read_cmd % devname
+            logging.info("Begin to write usb disk with dd cmd")
+            status, output = \
+                session.cmd_status_output(write_cmd, timeout=int(60))
+            if status != 0:
+                raise exceptions.TestFail("Failed to write usb disk with error: %s"
+                                          % output)
+            logging.info("Begin to read usb disk with dd cmd")
+            status, output = \
+                session.cmd_status_output(read_cmd, timeout=int(60))
+            if status != 0:
+                raise exceptions.TestFail("Failed to read usb disk with error: %s"
+                                          % output)
+        elif with_fs == "yes":
+            ranstr = utils_misc.generate_random_string(100)
+            write_cmd = "echo %s >> /media/testfile" % ranstr
+            read_cmd = "cat /media/testfile"
+            status, output = \
+                session.cmd_status_output(write_cmd, timeout=int(60))
+            if status != 0:
+                raise exceptions.TestFail("Write to file error: %s" % output)
+            status, output = \
+                session.cmd_status_output(read_cmd, timeout=int(60))
+            if status != 0:
+                raise exceptions.TestFail("Read file error: %s" % output)
+            if output.strip() != ranstr:
+                raise exceptions.TestFail("The content written to file has "
+                                          "changed,from: %s, to: %s" %
+                                          (ranstr, output.strip()))
+
+    def format_usb_disk(self, devname, session):
+
+        """
+        format usb disk
+        """
+        result = utils_misc\
+            .format_guest_disk(session,
```

Comment:
It's a func(), it looks like really wired.
You can write it such as:
utils_misc.format_guest_disk(session, devname,
                                                 "/media", 100, "xfs", "linux")